backend/tests/Feature/Api/V1/EngineeringDocumentTest.php

```
@@ -0,0 +1,239 @@
+
+    public function test_upload_for_a_missing_owner_is_rejected(): void
+    {
+        $this->actingAs($this->admin)
+            ->postJson('/api/v1/engineering-documents', $this->payload(['entity_id' => 999999]))
+            ->assertStatus(422)
+            ->assertJsonValidationErrors('entity_id');
+    }
+
+    public function test_operator_can_view_but_not_upload(): void
+    {
+        EngineeringDocument::factory()->create(['entity_type' => 'material', 'entity_id' => $this->material->id]);
```

PHPStan (2.2.6)

[error] Call to an undefined static method App\Models\EngineeringDocument::factory().


(staticMethod.notFound)